ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,387 @@
+module ManoptJuMPExt
+
+using Manopt
+using LinearAlgebra
+using JuMP: JuMP
+const MOI = JuMP.MOI
+using ManifoldsBase
+using ManifoldDiff
+
+function __init__()
+    # So that the user can use the convenient `Manopt.Optimizer`
+    setglobal!(Manopt, :Optimizer, Optimizer)
+    return nothing
+end
+
+"""
+    struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+        manifold::M
+    end
+
+Representation of points of `manifold` as a vector of `R^n` where `n` is
+`MOI.dimension(VectorizedManifold(manifold))`.
+"""
+struct VectorizedManifold{M<:ManifoldsBase.AbstractManifold} <: MOI.AbstractVectorSet
+    manifold::M
+end
+
+"""
+    MOI.dimension(set::VectorizedManifold)
+
+Return the dimension of the vectorized Manifold. As the MOI variables are real
+if the dimension is `n`, it means that the vectorization is a subset of `R^n`.
+"""
+function MOI.dimension(set::VectorizedManifold)
+    return prod(ManifoldsBase.representation_size(set.manifold))
+end
+
+"""
+    Manopt.Optimizer()
+
+Creates a new optimizer object.
+
+The minimization of a function `f(X)` of an an array `X[1:n1,1:n2,...]`
+over a manifold `M` starting at `X0`, can be modeled as follows:
+```julia
+using JuMP
+@variable(model, X[i1=1:n1,i2=1:n2,...] in M, start = X0[i1,i2,...])
```

Comment:
one should mention either in code or in the text before or after, that `model` is such an Optimizer`?

Reply:
Good point